OpenRA.Mods.Common/Traits/Sellable.cs

```diff
@@ -66,17 +67,33 @@ public void Sell(Actor self)
 			foreach (var ns in self.TraitsImplementing<INotifySold>())
 				ns.Selling(self);
 
-			if (!info.SkipMakeAnimation)
+			if (info.SkipMakeAnimation)
+				self.QueueActivity(false, new Sell(self));
+
+			PlaySellAnimations(self);
+		}
+
+		void PlaySellAnimations(Actor self)
+		{
+			var makeAnimations = self.TraitsImplementing<WithMakeAnimation>();
+			if (!makeAnimations.Any())
 			{
-				var makeAnimation = self.TraitOrDefault<WithMakeAnimation>();
-				if (makeAnimation != null)
+				// Let the sell happen even without WithMakeAnimation trait.
+				self.QueueActivity(false, new Sell(self));
+				return;
+			}
+
+			bool first = true;
+			foreach (var ma in makeAnimations)
+			{
+				if (first)
```

Comment:
Attaching this to the first defined animation will break selling if the standard make animation trait isn't first.  You can reproduce the issue by adding the following to the bottom of a structure:
```
	WithMakeAnimation@fake:
		DeployTypes: notreal
	-WithMakeAnimation:
	WithMakeAnimation:
```

Reply:
I suggest doing something like the following to make sure that the sell(/transform/etc) runs only after the last animation completes:
```csharp
var animationsActive = 0;
Action onAnimationComplete = () =>
{
	if (--animationsActive == 0)
		self.QueueActivity(false, new Sell(self));
};

foreach (var ma in makeAnimations)
{
	animationsActive++;
	ma.Reverse(self, onAnimationComplete);
}
```

However, something deeper in the make animation / sprite body code prevents the callback from running on all animations, and so this isn't enough to fix the bug by itself.

Reply:
Yes, we need to identify the longest animation, considering Tick parameter and the length of the frames. Does ORA engine have net animation length calculation already?

Reply:
It does not, and in general that isn't feasible to do (there are too many factors that would break such a calculation - overriding animations, pausing/cancelling).  This is why the callbacks exist.

Reply:
The above approach doesn't work reliably.

	foreach (var ma in makeAnimations)
			{
				if (ma.IsTraitEnabled())
				{
					activeAnimations++;
					ma.Reverse(self, onAnimationComplete);
				}
			}

ma.IsTraitEnabled works for make animation itself and there isn't a way to check if WSB is actually active.